### PR TITLE
adapter interface01

### PR DESCRIPTION
**Pull request by xiaoqin2012**
## Overview

Description:

Why should this be merged: 

Related issue(s) (if applicable): #<number>

## Checklist (Definition of Done):

- [ ] There are no TODOs left in the code
- [ ] [Coding conventions](https://github.com/CorfuDB/CorfuDB/wiki/Corfu-Style-Guidelines) (e.g. for logging, unit tests) have been followed
- [ ] Change is covered by automated tests
- [ ] Public API has Javadoc

### CODE REVIEW

File: infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/plugins/CorfuReplicationSiteManagerAdapter.java [hunk 1]
```diff
@@ -5,47 +5,26 @@
...
-        return corfuReplicationDiscoveryService.queryReplicationStatus();
-    }
-
-    //TODO: handle the case that queryTopologyConfig return an exception.
-    public abstract TopologyConfigurationMsg queryTopologyConfig();
-
-    public abstract void start();
-
-    public abstract void shutdown();
+public interface CorfuReplicationSiteManagerAdapter {
```
Comment (annym) [inline infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/plugins/CorfuReplicationSiteManagerAdapter.java:8]: Site -> Cluster

File: infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/plugins/CorfuReplicationSiteManagerAdapter.java [hunk 2]
```diff
@@ -5,47 +5,26 @@
...
+
+    void setCorfuReplicationDiscoveryService(CorfuReplicationDiscoveryServiceAdapter corfuReplicationDiscoveryService);
```
Comment (annym) [inline infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/plugins/CorfuReplicationSiteManagerAdapter.java:10]: missing accessors.

File: infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/plugins/CorfuReplicationSiteManagerAdapter.java [hunk 3]
```diff
@@ -5,47 +5,26 @@
...
+
+    // This is the currentTopology cached at the local node.
+    public TopologyConfigurationMsg getTopologyConfig();
+
+    // This will talk to the real Site Manager and get the most current topology.
+    TopologyConfigurationMsg queryTopologyConfig();
+
+    // This is called when get a notification of site config change.
+    void updateTopologyConfig(TopologyConfigurationMsg newSiteConfigMsg);
+
```
Comment (annym) [inline infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/plugins/CorfuReplicationSiteManagerAdapter.java:22]: we are missing accessors and comments to each method of the interface.

File: infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/plugins/CorfuReplicationSiteManagerAdapter.java [hunk 4]
```diff
@@ -5,47 +5,26 @@
...
+    void start();
+
+    void shutdown();
+
+    void prepareSiteRoleChange();
```
Comment (annym) [inline infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/plugins/CorfuReplicationSiteManagerAdapter.java:27]: site -> cluster

File: infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/plugins/CorfuReplicationSiteManagerBaseAdapter.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+import lombok.Getter;
+
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.infrastructure.logreplication.infrastructure.CorfuReplicationDiscoveryServiceAdapter;
+import org.corfudb.infrastructure.logreplication.proto.LogReplicationClusterInfo.TopologyConfigurationMsg;
+
+/***
+ * This the base class for CorfuReplicationSiteManager, and have the basic functionalities implemented.
+ * Any SiteMangerImplementation should extends this class or implements the interface.
```
Comment (annym) [inline infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/plugins/CorfuReplicationSiteManagerBaseAdapter.java:11]: typo -> extend

File: infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/plugins/CorfuReplicationSiteManagerBaseAdapter.java [hunk 6]
```diff
@@ -0,0 +1,52 @@
+package org.corfudb.infrastructure.logreplication.infrastructure.plugins;
+
+import lombok.Getter;
+
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.infrastructure.logreplication.infrastructure.CorfuReplicationDiscoveryServiceAdapter;
+import org.corfudb.infrastructure.logreplication.proto.LogReplicationClusterInfo.TopologyConfigurationMsg;
+
+/***
+ * This the base class for CorfuReplicationSiteManager, and have the basic functionalities implemented.
```
Comment (annym) [inline infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/plugins/CorfuReplicationSiteManagerBaseAdapter.java:10]: typo -> this is

File: infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/plugins/CorfuReplicationSiteManagerAdapter.java (hunk 2 above)
Comment (xiaoqin2012) [reply infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/plugins/CorfuReplicationSiteManagerAdapter.java:10]: It is not supposed to be accessible.

File: infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/plugins/CorfuReplicationSiteManagerAdapter.java (hunk 1 above)
Comment (xiaoqin2012) [reply infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/plugins/CorfuReplicationSiteManagerAdapter.java:8]: I am really confused with all the new naming. Corfu is a cluster, but we don't name anything there as cluster. If we follow corfu's naming, we should not introduce cluster in the naming. 
When naming as ClusterManager, it gives the conception of it is manager the whole cluster, but it only gets information for this local node to process.

Comment (annym) [reply infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/plugins/CorfuReplicationSiteManagerAdapter.java:14]: no Corfu is not a Cluster. Corfu is the friendly name of our Datastore. 
I was referring to name it CorfuReplicationClusterManagerAdapter... We agreed that Site is not a 'Corfu' terminology, we're broader in scope than a Site, we talk in terms of Clusters (i.e., Corfu Clusters). 
If you still have questions we can discuss it offline.

File: infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/plugins/CorfuReplicationSiteManagerAdapter.java (hunk 2 above)
Comment (annym) [reply infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/plugins/CorfuReplicationSiteManagerAdapter.java]: then the accessor is private. It's to be consistent with Corfu's coding style.

**Comment by hisundar** [comment]
Hi Xiaoqin, I am working on integrating with this change, but now facing issues after the recent merge .
Could you please rebase this on top of the latest log-replication-feature branch so I can pick it up on next drop?

**Comment by xiaoqin2012** [comment]
I will look into it.
________________________________

**Pull request closed without merging**